src/Kunstmaan/NodeBundle/Helper/Services/PageCreatorService.php

```diff
@@ -32,37 +35,59 @@ class PageCreatorService
      */
     protected $userEntityClass;
 
+    /**
+     * @var HasNodeInterface $pageTypeInstance
+     */
+    private $pageTypeInstance;
 
-    public function setEntityManager($entityManager)
-    {
-        $this->entityManager = $entityManager;
-    }
+    /**
+     * @var Node
+     */
+    private $rootNode = null;
 
-    public function setACLPermissionCreatorService($aclPermissionCreatorService)
-    {
-        $this->aclPermissionCreatorService = $aclPermissionCreatorService;
-    }
+    /**
+     * @var \Kunstmaan\SeoBundle\Entity\Seo|null $seo
+     */
+    private $seo;
 
-    public function setUserEntityClass($userEntityClass)
-    {
-        $this->userEntityClass = $userEntityClass;
-    }
+    /**
+     * @var User $creator
+     */
+    private $creator;
 
     /**
-     * Sets the Container. This is still here for backwards compatibility.
-     *
-     * The ContainerAwareInterface has been removed so the container won't be injected automatically.
-     * This function is just there for code that calls it manually.
-     *
-     * @param ContainerInterface $container A ContainerInterface instance.
-     *
-     * @api
+     * @var NodeRepository $nodeRepo
      */
-    public function setContainer(ContainerInterface $container = null)
+    private $nodeRepo;
+
+    /**
+     * @var NodeTranslationRepository $translationsRepo
+     */
+    private $translationsRepo;
+
+    /**
+     * @var array $options
+     */
+    private $options;
+
+    /**
+     * @var mixed|null $parent
+     */
+    private $parent;
+
+    /**
+     * PageCreatorService constructor.
+     * @param EntityManager $em
+     * @param ACLPermissionCreatorService $aclPermissionCreatorService
+     * @param $userClass
+     */
+    public function __construct(EntityManager $em, ACLPermissionCreatorService $aclPermissionCreatorService, NodeRepository $nodeRepo, NodeTranslationRepository $translationRepo, $userClass)
```

Comment:
If You have much arguments in function __constructor should looked like this:
```
    public function __construct(
        EntityManager $em,
        ACLPermissionCreatorService $aclPermissionCreatorService,
        NodeRepository $nodeRepo,
        NodeTranslationRepository $translationRepo,
        $userClass
    )
    {
        $this->entityManager = $em;
        $this->aclPermissionCreatorService = $aclPermissionCreatorService;
        $this->userEntityClass = $userClass;
        $this->nodeRepo = $nodeRepo;
        $this->translationsRepo = $translationRepo;
    }
```
It is more readable. Also fill the docblock.

Reply:
Or if it's getting to long, maybe we should create a helper service with some functions in it.